admiral/pkg/clusters/serviceentry_test.go

```diff
@@ -268,17 +278,78 @@ import (
 //	}
 //}
 //
-func buildFakeConfigMapFromAddressStore(addressStore *ServiceEntryAddressStore, resourceVersion string) *v1.ConfigMap{
-	bytes,_ := yaml.Marshal(addressStore)
+func buildFakeConfigMapFromAddressStore(addressStore *ServiceEntryAddressStore, resourceVersion string) *v1.ConfigMap {
+	bytes, _ := yaml.Marshal(addressStore)
 
 	cm := v1.ConfigMap{
 		Data: map[string]string{"serviceEntryAddressStore": string(bytes)},
 	}
-	cm.Name="se-address-configmap"
-	cm.Namespace="admiral-remote-ctx"
-	cm.ResourceVersion=resourceVersion
+	cm.Name = "se-address-configmap"
+	cm.Namespace = "admiral-remote-ctx"
+	cm.ResourceVersion = resourceVersion
 	return &cm
 }
+
+func TestModifyNonExistingSidecarForLocalClusterCommunication(t *testing.T) {
+	config, error := clientcmd.BuildConfigFromFlags("", "../test/resources/admins@fake-cluster.k8s.local")
+
+	if error != nil {
+		t.Errorf("%v", error)
+	}
+	stop := make(chan struct{})
+	handler := test.MockSidecarHandler{}
+
+	sidecarController, _ := istio.NewSidecarController(stop, &handler, config, time.Duration(1000))
+
+	remoteController := &RemoteController{SidecarController: sidecarController}
+
+	modifySidecarForLocalClusterCommunication("test-sidecar-namespace", "test-dependency-namespace", "test-local-fqdn", remoteController)
+
+	sidecarObj, _ := sidecarController.IstioClient.NetworkingV1alpha3().Sidecars("test-sidecar-namespace").Get(common.GetWorkloadSidecarName(), v12.GetOptions{})
+
+	if sidecarObj == nil || sidecarObj.Spec.Egress != nil {
+		t.Fail()
+	}
+}
+
+func TestModifyExistingSidecarForLocalClusterCommunication(t *testing.T) {
+
+	sidecarController := &istio.SidecarController{}
+	sidecarController.IstioClient = istiofake.NewSimpleClientset()
+
+	remoteController := &RemoteController{}
+	remoteController.SidecarController = sidecarController
+
+	existingSidecarObj := &v1alpha3.Sidecar{}
+	existingSidecarObj.ObjectMeta.Namespace = "test-sidecar-namespace"
+	existingSidecarObj.ObjectMeta.Name = "default"
+
+	istioEgress := istionetworkingv1alpha3.IstioEgressListener{
+		Hosts: []string{"test-host"},
+	}
+
+	existingSidecarObj.Spec = istionetworkingv1alpha3.Sidecar{
+		Egress: []*istionetworkingv1alpha3.IstioEgressListener{&istioEgress},
+	}
+
+	createdSidecar, _ := sidecarController.IstioClient.NetworkingV1alpha3().Sidecars("test-sidecar-namespace").Create(existingSidecarObj)
+
+	if createdSidecar != nil {
+
+		modifySidecarForLocalClusterCommunication("test-sidecar-namespace", "test-dependency-namespace", "test-local-fqdn", remoteController)
+
+		updatedSidecar, error := sidecarController.IstioClient.NetworkingV1alpha3().Sidecars("test-sidecar-namespace").Get("default", v12.GetOptions{})
+
+		if error != nil || updatedSidecar == nil {
+			t.Fail()
+		}
+
+		if len(updatedSidecar.Spec.Egress[0].Hosts) != 2 {
```

Comment:
Same as above - in my opinion the tests are the most valuable when we're asserting for exactly what's needed. If there's some sidecar egress formatting bug introduced later, we want these tests to catch it

Reply:
Agreed, modified the test